### DISCUSSION TITLE
Localize a value of attribute

### DISCUSSION THREAD

**Discussion by KaiHuaDou** [dotnet/csharplang#7378] [Q&A] [answered]
```csharp
    [Option('i', "input", Required = true, HelpText = "My Help Text")]
    public string Input { get; set; }
```

Is there any solution to localize the string "My Help Text"?

**Comment by huoyaoyuan** [comment]
The common practice is to encode a key in attribute/code, then implement lookup from localized resource when displaying.
Is this related to [System.CommandLine](https://github.com/dotnet/command-line-api)?

**Reply by KaiHuaDou** [reply]
I use `CommandLineParser`, So  it means I should set the value of the attributte in runtime with reflection?

**Reply by huoyaoyuan** [reply, marked as answer]
Attribute data can't be set via reflection. You can manually create the data structure parsed from attribute, or create a feature request about localization.
